docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,164 @@
+
+Assessing model robustness is the tricky part. Standard benchmarks do not apply: what you are going to measure is actually a loss of performance as the specialization through retraining only results in a loss of general performance.
+
+As far as accuracy goes, the most straightforward way is to compare the model output and the original documents. It works obviously best if you are already clarifying the attribution of individual statements to specific documents through generated references. For this process you can use an LLM-as-a-judge or, if you provide actual quotes, a straight comparison with the original text (as an additional confusing factor, this is called a “text alignment” algorithm, but nothing to do with fine-tuning or RLHF).
+
+### 4.3 Purpose of fine-tuning: relieving data frictions
+A RAG system is made primarily for its users. Hence one of the biggest issues of using a generalist model like GPT-3.5/4 on a long term basis: you have an audience alignment gap. This is even more striking if you’re working in another language than English (like myself): ChatGPT feels immediately “alien” with weird twists of languages that are not to be found in standard French.
+
+Reshaping style is the most shocking impact of fine-tuning. A few months ago I released an extreme version of user-LLM communication: MonadGPT, an instruction model trained on 17th century English. What took me the longest was to come up with the most fitting communication scheme: train the models on a couple of answers from historical texts and generated questions in contemporary English, simply because I should absolutely not expect the users to mimic 17th century language. Users should not have to mimic the style of any document resources either. 
+
+Similarly to MonadGPT you should look for the following alignment design:
+* Generated queries should be as close as possible as the real queries sent by users, which means likely purposefully generating “bad” data. In my current instruction dataset, most queries have no punctuation, some are in all caps or missing words, with typos.
+* Generated answers have to convey the general “brand” style of your system. Do not hesitate to re-generate the answer, reformulate the existing one, or even rewrite them manually.
+
+Relatedly, you have to anticipate the “persona” of your model. Whether you like it or not, users of interactive systems will ask meta-referential questions: who are? What can you do? Many advanced LLM systems overlooked this aspect to their detriment (with new models trained from scratch erroneously stating they are ChatGPT, since this is the default AI persona in much of the training data these days). I recommend here to combine both prepared answers in the instruction dataset with a series of “biographical” documents about the model in the RAG dataset.
+
```

Comment:
"prepared answers in the instruction dataset"
"'biographical' documents about the model in the RAG dataset"
Are the instruction dataset and RAG dataset different?

Reply:
Yes basically. RAG dataset is the vector store. This is roughly a redondancy strategy to reinforce the overall "persona" effect: model is fine-tuned on instruction reflecting that + has access in inference time to stored information about itself.